compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/FakeInliningLocalVariablesLowering.kt

```diff
@@ -49,10 +48,14 @@ internal class FakeInliningLocalVariablesLowering(val context: JvmBackendContext
         // Do not record inline function callable reference parameters. They will not be used.
     }
 
-    override fun handleInlineFunctionLambdaParam(lambda: IrFunction, callee: IrFunction, callSite: IrDeclaration?) {
+    override fun handleInlineFunctionLambdaParam(lambdaReference: IrFunctionReference, callee: IrFunction, callSite: IrDeclaration?) {
         // Do not record lambda parameters. Instead deal with them now.
+        val lambda = lambdaReference.symbol.owner
         if (lambda.origin == IrDeclarationOrigin.LOCAL_FUNCTION_FOR_LAMBDA) {
-            lambda.addFakeInliningLocalVariablesForArguments(callee)
+            val argumentToFunctionName = context.methodSignatureMapper.mapFunctionName(callee)
+            val lambdaReferenceName = context.getLocalClassType(lambdaReference)!!.internalName.substringAfterLast("/")
```

Comment:
Does this mean that the names used for the fake local variables are different from the names in the actual class files? Or was that the case before? Should there be a corresponding change in the way we name the actual elements in the output? I would find it very confusing if the names referred to in the fake local variables were different from what is in the class file. I can't figure out if this change is fixing that or making it a problem?

Reply:
The names for local classes are also of the form `enclosingName$n`, where the enclosing name is of the form `enclosingType$enclosingFunction` for lambdas defined inside of a function. The main difference to the previous behavior is the `$n` suffix, which distinguishes the lambda from all other anonymous classes in the current file.

Reply:
Makes sense, thanks Steven! So the point here is that all the references to named elements that are in the output class files have the right names. Only the numbering of the lambdas which are no in the actual output is changed here. And for full context these names seem to be used by the step-in handler in the Intellij debugger.